<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Camera topic  -->
    <arg name="image_topic"	                default="/camera/color/image_raw"/>
    <arg name="use_depth"                   default="true"/> <!-- If false, only use RGB and distance is 0mm-->
    <arg name="depth_topic"	                default="/camera/aligned_depth_to_color/image_raw"/>
    
    <!-- Published topics -->
    <arg name="publish_image"	            default="true"/>
    <arg name="detected_objects"       default="detected_objects"/>
    <arg name="detected_image"        default="detected_image"/>

    <!-- Detection model onfiguration -->
    <arg name="weights_name"	            default="yolov5/weights/yolov5s.pt"/>
    <arg name="img_size"                    default="640"/> <!-- input size-->
    <arg name="confidence"                  default="0.4"/>
    <arg name="nms_th"                      default="0.5"/>
    <arg name="class_name"                  default='person'/> <!-- COCO index name, 'None': all class detect-->
    <arg name="augment"                     default="false"/> <!-- data augmentation-->
    <arg name="agnostic_nms"                default="false"/> <!-- data augmentation-->
    
    <!-- <arg name="class_name"                  default='0, 1, 2'/> -->
    <!-- <arg name="class_name"                  default='person'/> -->

    <!-- if you don't have GPU input the value 'cpu' -->
    <arg name="gpu_id"                      default="0"/>
   
    <!-- Node -->
    <node name="detector_yolo_node" pkg="detector" type="detect_yolo.py" output="screen" respawn="true">
        <param name="image_topic"               value="$(arg image_topic)"/>
        <param name="use_depth"                 value="$(arg use_depth)"/>
        <param name="depth_topic"               value="$(arg depth_topic)"/>
        <param name="publish_image"             value="$(arg publish_image)"/>
        <param name="detected_objects"     value="$(arg detected_objects)"/>
        <param name="detected_image"      value="$(arg detected_image)"/>
        <param name="weights_name"	            value="yolov5/weights/yolov5s.pt"/>
        <param name="img_size"                    value="480"/> <!-- input size-->
        <param name="confidence"                  value="0.4"/>
        <param name="nms_th"                      value="0.5"/>
        <param name="class_name"                  value='None'/> <!-- COCO index name, 'None': all class detect-->
        <param name="augment"                     value="false"/> <!-- data augmentation-->
        <param name="agnostic_nms"                value="false"/> <!-- data augmentation-->
        <param name="class_name"                 value="$(arg class_name)"/>
        <param name="gpu_id"                    value="$(arg gpu_id)"/>
        </node>
</launch>
